dashboard/cron.py

```diff
@@ -610,12 +609,16 @@ def do(self):
             logger.info("************ total status=" + status + "/n/n")
             return (status,)
 
+        # Lock in valid course IDs that data will be pulled for.
+        course_df: pd.DataFrame = course_verification.course_data
+        self.valid_locked_course_ids: List[int] = course_df['id'].to_list()
```

Comment:
It may not be a best practice to initialize instance variables outside a constructor, but it seemed less messy than overriding the `__init__` from `CronJobBase` See https://stackoverflow.com/questions/19284857/instance-attribute-attribute-name-defined-outside-init I'll be looking into this more.

Reply:
I believe you just can define an __init__ here and call 
`super().__init__()`

You could also put it in the do() method as that will always be called first.

Reply:
I think you need to pass through the constructor parameters to the `super().__init()` call. Not sure what they are, I'd have to look into the code, since it's not documented. Even if I did figure it out, I'd be concerned that might easily be broken by updates to that package (though there haven't been many lately).

Reply:
Yeah I didn't see this one having any parameters other than self. 
https://github.com/Tivix/django-cron/blob/7fa5c2d2835520d8c920f3c103937ab292d9e545/django_cron/__init__.py#L52

Reply:
Hm, nice, I'll try that. Thanks for doing the work for me!

Reply:
No problem, just working my crunch time developer support shift! :)

Reply:
Done! Thought about initializing as `None`, but it throws `mypy` error(s) and I thought it was kinda much to check everywhere whether it was a list. `mypy` seems to be happy with this. See https://stackoverflow.com/questions/60925137/using-mypy-with-with-lazy-initialization-of-instance-attributes